nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/SecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,191 @@
+     * does not exist, this will generate a key pair and self-signed certificate, generate the associated keystore
+     * and truststore and write them to disk under the configured filepaths, generate a secure random keystore password
+     * and truststore password, and write these to the nifi.properties file.
+     * @param nifiPropertiesFilename The filename of the nifi.properties file
+     * @param cmdLogger The bootstrap logger
+     * @throws IOException can be thrown when writing keystores to disk
+     * @throws RuntimeException indicates a security exception while generating keystores
+     */
+    public static void configureSecureNiFiProperties(String nifiPropertiesFilename, Logger cmdLogger) throws IOException, RuntimeException {
+                final File propertiesFile = new File(nifiPropertiesFilename);
+        final Properties nifiProperties = loadProperties(propertiesFile);
+
+        if (!nifiProperties.getProperty(NiFiProperties.WEB_HTTPS_PORT, "").isEmpty()) {
+            String keystorePath = nifiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE, "");
+            String truststorePath = nifiProperties.getProperty(NiFiProperties.SECURITY_TRUSTSTORE, "");
+            if (!keystorePath.isEmpty() && !truststorePath.isEmpty()) {
+                File keystore = new File(keystorePath);
+                File truststore = new File(truststorePath);
+                if (!keystore.exists() && !truststore.exists()) {
+                    TlsConfiguration tlsConfiguration = null;
+                    cmdLogger.info("Generating default self-signed certificates, keystore and truststore for secure Apache NiFi configuration.  " +
+                            "This certificate will expire in {} days, and it is recommended to acquire your own certificates in order to properly secure Apache NiFi.", CERT_DURATION_DAYS);
+                    try {
+                        Pair<String[], String[]> subjectAlternativeNames = getSubjectAlternativeNames(nifiProperties);
+                        tlsConfiguration = KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore(StandardTlsConfiguration
+                                .fromNiFiProperties(nifiProperties), CERT_DURATION_DAYS, subjectAlternativeNames.getLeft(),
+                                subjectAlternativeNames.getRight());
+                    } catch (IOException e) {
+                        cmdLogger.error("Encountered an I/O exception while generating secure Apache NiFi configuration.", e);
+                        throw e;
+                    } catch (GeneralSecurityException e) {
+                        cmdLogger.error("Encountered a security exception while generating secure Apache NiFi configuration.", e);
+                        throw new RuntimeException(e);
+                    }
+
+                    // Move over the new stores from temp dir
+                    Files.move(new File(tlsConfiguration.getKeystorePath()).toPath(), new File(keystorePath).toPath(),
```

Comment:
Is it possible to use `Paths.get()` instead of `new File().toPath()`?  There may be tradeoffs to this approach depending on relative file path handling, but it may also be worth creating a private method called `getPath(String filePath)` to streamline creation of `Path` objects.